pages/studentPortal/quizcomponent/FirstProgrammingLanguageQuiz.tsx

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from "react";
+import LangResults from "../../../components/resources/quizzes/langQuiz/LangResults";
+import BluePrint from "../../../components/resources/quizzes/shared/BluePrint";
+import SkillSelections from "../../../components/resources/quizzes/shared/SkillSelections";
+import StartQuiz from "../../../components/resources/quizzes/shared/StartQuiz";
+export enum Stage {
+  START,
+  LEARNCODING,
+  WORKFIELDS,
+  BUILDTARGET,
+  BLUEPRINT,
+  RESULTS,
+}
+
+type QuizSelection = {
+  name: string;
+  result?: string;
+  weight?: number;
+};
+
+const FirstProgrammingLanguageQuiz = () => {
+  const quiz: QuizSelection[][] = [
+    [
+      { name: "Learn an in-demand skill" },
+      { name: "Work in tech" },
+      { name: "Build programs for fun" },
+      { name: "I run a startup" },
+      { name: "Become a professional developer" },
+      { name: "Communicate better with technical coworkers" },
+      { name: "Start a side hustle" },
+    ],
+    [
+      { name: "Front end development", result: "JavaScript", weight: 1 },
+      { name: "Back end development", result: "JavaScript", weight: 1 },
+      { name: "Game development", result: "JavaScript", weight: 1 },
+      { name: "Mobile development", result: "JavaScript", weight: 1 },
+      { name: "Data analytics", result: "Python", weight: 1 },
+      { name: "Product management", result: "JavaScript", weight: 1 },
+      { name: "Digital marketing", result: "JavaScript", weight: 1 },
+      { name: "UX/UI design", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
+    ],
+    [
+      { name: "Websites", result: "JavaScript", weight: 1 },
+      { name: "Storefront", result: "HTML/CSS", weight: 1 },
+      { name: "Mobile apps", result: "JavaScript", weight: 1 },
+      { name: "Games", result: "JavaScript", weight: 1 },
+      { name: "Email", result: "HTML/CSS", weight: 1 },
+      { name: "Tools to automate your life", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
+    ],
+  ];
+
+  const pageOptions = quiz.map((pages) => {
+    return pages.map((options) => {
+      return options.name;
+    });
+  });
+
+  type PageContent = {
+    title: string;
+    body: string;
+  };
+
+  const quizQuestions: PageContent[] = [
```

Comment:
This is already in a much better place since when you initially started, so great work on cleaning up this code already.

I want to point out some pieces that are confusing.

You have two variables called quiz and quizQuestions, I can't tell which one is supposed to be the single source of truth for quiz state, and how the variables are related. 

The names of your variables don't really match the name of the type either.

What if you created a type called `QuizData` and all the quiz questions and all the title and body texts were captured inside of that object. 

That QuizData object would be the single source of truth for the quiz. You're basically combining the quiz and quizQuestions into a single variable.

I would differentiate between QuizData and QuizState. QuizData just has the info about the quiz questions and weight/title/body text. QuizData has no idea about what the user has selected.

QuizState would have the quiz data inside of it, but also hold information about what the user has currently selected.

I'd also be careful to differentiate between a quiz option and a quiz selection.